### DISCUSSION TITLE
Prefect with DaskExecutor with adapt not running in parallel

### DISCUSSION THREAD

**Discussion by porbmv83** [PrefectHQ/prefect#5440] [Q&A]
Hello,

I'm doing some tests running prefect with DaskExecutor in my AKS cluster using a custom docker image.
Please check the following flows:

- Flow running with adaptive number of workers [https://github.com/porbmv83/prefect_test/blob/main/dask_spre.py](url) - It takes 4:30 minutes and it doesn't seem to be parallelizing the execution using all the working nodes.
![image](https://user-images.githubusercontent.com/93665757/153872665-7def6fbc-28d7-40ed-b223-28e2a8639f54.png)

- Flow running with static workers [https://github.com/porbmv83/prefect_test/blob/main/dask_spre_no_autoscale.py](url) - it takes less than 53 seconds
![image](https://user-images.githubusercontent.com/93665757/153872878-056df1e4-4f82-45bc-9238-742393803d90.png)

Am I doing something wrong?

Thanks
Bruno

**Comment by BitTheByte** [comment]
Same problem here

**Comment by anna-geller** [comment]
So the Dask cluster on K8s has been created successfully, but the task runs somehow don't get submitted to the Dask cluster?

To see where the issue is, you can try this flow example first to see if you get parallelism with just LocalDaskExecutor.

```python
from prefect import task, Flow
from prefect.executors import LocalDaskExecutor


@task
def generate_random_numbers():
    return list(range(1, 200))


@task
def add_one(x):
    return x + 1


@task(log_stdout=True)
def print_results(res):
    print(res)


with Flow("mapping", executor=LocalDaskExecutor()) as flow:
    numbers = generate_random_numbers()
    result = add_one.map(numbers)
    print_results(result)
```

If this works well on AKS, you can then try with a local Dask cluster (no KubeCluster yet).
```python
with Flow("mapping", executor=DaskExecutor()) as flow:
```
If this worked too, you can then run the same simple flow from here on KubeCluster but with the most vanilla setup - no custom image, just the default configuration.

Then add only the image:
```python
with Flow(
    FLOW_NAME,
    storage=STORAGE,
    executor=DaskExecutor(
        cluster_class=lambda: KubeCluster(make_pod_spec(image=prefect.context.image)),
        adapt_kwargs={"minimum": 2, "maximum": 3},
    ),
    run_config=KubernetesRun(
        labels=["dask"],
        image="annageller/prefect-dask-k8s:latest", # just an example image, you can use your ACR image
    ),
) as flow:
```

If that worked, then check with a full KubeCluster pod spec. 

By doing it step by step, you can easier determine where the process fails.  

You can also check a full working example for a KubeCluster with a custom image here: https://github.com/anna-geller/packaging-prefect-flows/tree/master/ephemeral_dask_cluster

**Comment by zanieb** [comment]
Perhaps this is related to https://github.com/PrefectHQ/prefect/issues/5549 ? What versions of distributed and prefect are you using?

**Comment by BitTheByte** [comment]
@madkinsz  I don't think it has something to do with https://github.com/PrefectHQ/prefect/issues/5549 
```
prefect: 1.2.0
distributed: 2022.4.1
```
from what I read, prefect 1.0 has a limitation that it could only parallelize one mapping task each maybe that's the cause?
